src/main/java/com/webapp/TextBook/viewModel/apiViewModel/StudentInfo.java

```diff
@@ -0,0 +1,175 @@
+     */
+    public String termCode;
+    /***
+     * Constructors
+     *         1. Blank
+     *         2. Takes all inputs
+     */
+    public StudentInfo(){}
+    public StudentInfo(String id, String termCode){
+        this.id = id;
+        this.termCode = termCode;
+
+    }
+
+    /***
+     * All neccessary getters and setters for non-final values
+     */
+
+    public String getId() {
+        return id;
+    }
+
+    public void setId(String id) {
+        this.id = id;
+    }
+
+    public String getTermCode() {
+        return termCode;
+    }
+
+    public void setTermCode(String termCode) {
+        this.termCode = termCode;
+    }
+    /***
+     *  Implements the needed method. Refer to LoginUserInfo for context.
+     */
+    /*
+    public @NotNull Optional<StudentInfo> createApiViewModelFromJson(@NotNull JSONObject jsonObject, Supplier<StudentInfo> initialInstantiation) {
+        final StudentInfo student = new StudentInfo();
+
+        // try-catch for JSON key errors in parsing partial user definition
+        try{
+
+            student.setId(jsonObject.getString(StudentInfo.NOMINAL_ID));
+
+            student.setTermCode(jsonObject.getString(StudentInfo.NOMINAL_TERM_CODE));
+
+            return Optional.of(student);
+
+        }
+        catch(JSONException jsonException){
+            // logs (prints now) exception
+            System.out.println(jsonException.getMessage());
+
+            return Optional.empty();
+        }
+    }
+
+     */
+
+    /**
+     * <p>
+     *     From the JSONObject (should be JSON composition/reflection of StudentInfo's state)
+     *     and an optional Supplier, render and sets a StudentInfo object to the reflective/corresponding
+     *     state within the JSONObject.
+     * </p>
+     * @param jsonObject JSONObject which is the composition of the state that defines a StudentInfo
+     * @param studentInfoSupplier(type param: StudentInfo) optional function to invoke to supplement
+     *                              a StudentInfo to use in creation and state setting.
+     * @return Optional(type param: StudentInfo) holding the ApiViewModel object. Will be empty if
+     * a) JSONObject does not hold the necessary state for a full state setting
+     * b) Supplier(type param: StudentInfo) throws an internal error
+     */
+    public static @NotNull Optional<StudentInfo> createApiFromJson(
+            @NotNull final JSONObject jsonObject,
+            @Nullable final Supplier<StudentInfo> studentInfoSupplier){
+
+        // creates a local supplier based on the predicate if the given supplier is null or not
+        // use the blank constructor as a default
+        final Supplier<StudentInfo> localLoginUserInfoSupplier = studentInfoSupplier == null ?
+                StudentInfo::new :
+                studentInfoSupplier;
+
+        // invokes parent's static helper to set the state for a LoginUserInfo
```

Comment:
not login user info
student info

Reply:
https://github.com/NicholasPierce1/TextBookServices/commit/8262f172f480d65bb3353bbb2d213e3e94f27951